ui/src/containers/ConfigScreen/index.tsx

```diff
@@ -133,14 +137,167 @@ const ConfigScreen: React.FC = function () {
     setModalOpen(false);
   };
 
+  const checkIsDefaultInitial = (configurationData: any) => {
+    const { default_multi_config_key } = configurationData;
+    if (default_multi_config_key === "") {
+      return true;
+    }
+    return false;
+  };
+
+  const checkValidity = async (
+    configurationData: { [x: string]: any; multi_config_keys?: any },
+    serverConfiguration: { [x: string]: any; multi_config_keys?: any },
+    validateMultiConfigKeysByApi: any,
+    validateOtherKeysByApi: any
+  ) => {
+    const checkMultiConfigKeys = (multiConfigKeys: any) => {
+      const invalidKeys: any = {};
+
+      Object.entries(multiConfigKeys || {})?.forEach(([configKey, config]) => {
+        Object.entries(config || {})?.forEach(([key, value]) => {
+          if (typeof value === "string" && value?.trim() === "") {
+            if (!invalidKeys?.[configKey]) {
+              invalidKeys[configKey] = [];
+            }
+            invalidKeys?.[configKey]?.push(key);
+          }
+        });
+      });
+
+      return invalidKeys;
+    };
+
+    const checkOtherKeys = (keys: any) => {
+      const invalidKeys: string[] = [];
+
+      Object.entries(keys || {})?.forEach(([key, value]) => {
+        if (typeof value === "string" && value?.trim() === "") {
+          invalidKeys?.push(key);
+        }
+      });
+
+      return invalidKeys;
+    };
+
+    let normalInvalidKeys = {};
+    let serverNormalInvalidKeys = {};
+
+    if (!validateMultiConfigKeysByApi) {
+      normalInvalidKeys = checkMultiConfigKeys(
+        configurationData?.multi_config_keys
+      );
+      serverNormalInvalidKeys = checkMultiConfigKeys(
+        serverConfiguration?.multi_config_keys
+      );
+    }
+
+    const normalInvalidKeysList = [
+      ...Object.entries(normalInvalidKeys)
```

Comment:
...Object.entries(normalInvalidKeys)
        ?.filter(([keys]) => keys?.length)
        ?.map(([source, keys]) => ({ source, keys })),
      ...Object.entries(serverNormalInvalidKeys)
        ?.filter(([keys]) => keys?.length)
        ?.map(([source, keys]) => ({ source, keys })),

Reply:
Unsafe usage of optional chaining. If it short-circuits with 'undefined' the evaluation will throw TypeError